src/core/createCookie.js

```diff
@@ -108,3 +119,11 @@ const createCookie = (componentNamespace, propertyId, cookieDomain = "") => {
 };
 
 export default createCookie;
+
+// #if _TEST
```

Comment:
You could avoid this by adding this as a parameter to the createCookieProxy function.  Then you would define the cookieProxyByPropertyId object in main.js.

Reply:
I'm not sure how I feel about this. In some ways, I like that it would remove the test-specific code, but on the other hand, `cookieProxyByPropertyId` feels like an implementation detail that shouldn’t be exposed to or modified by the consumer of the module. I'll bring it up in chat, because I'd like to see how others feel about it.

Reply:
I don't mind the `_TEST`; I see it as just another tool to help us test, similar to the build tools and plugins. And it makes it clear to understand that this portion is for testing purposes only; plus it looks like you are using it in multiple places. (To clear the top level domain cache)

I'd rather do that, than leaking the proxy stuff outside of this module.

Reply:
I think of it as the createCookie function having a dependency on a cache.  As it is in this PR, the cache is a hidden global variable.  In general explicit dependencies are better than hidden dependencies, and global state should be avoided. In this case, the test logic is fairly benign, but `#if _TEST` could quickly get out of hand to the point where the system behaves differently when under test.  

I'm good either way, and understand that everyone prefers testing in different ways.  I'm just pointing out that the `#if _TEST` could be avoided in this case.

Reply:
@jonsnyder I might have misunderstood your original comment. I absolutely agree that the system should behave the same way whether it's in test mode or not, and we definitely don't want hidden dependencies. 

DI is usually the answer to every similar situation; so the question is, are `proxy` in this case, and `cached topleveldomain` in the Top Level Domain module dependencies? Maybe. And maybe we have a `createMemoizer` or something that we re-use to inject modules in need of caching with a generic memoizer.

Reply:
Yes, I would make a similar comment on the getTopLevelCookieDomain function.  Expanding on what you said @jfkhoury, we could remove the caching/lookup logic from these functions, so they compute the values every time.  Then we could add a utility to memoize the function, and pass the memoized function to those that need it.

I misspoke before; these aren't really global variables.  They don't get defined on window, but it is global state.

getTopLevelCookieDomain should return the same value for any instance of alloy, so we need to have global state somewhere.  My preference is to explicitly have global state defined at the highest level (i.e. main.js).  The benefit of avoiding global state in these functions is that then they can be unit tested more easily.  I see main.js as containing glue code to wire up all the dependencies and avoiding logic.  Then main.js is tested using functional/integration tests to make sure everything is wired up properly and works together.

Reply:
Thanks for the feedback. I'm going to play around with these ideas and open in a new pull request if it shakes out well.